samples/subsys/usb/hid-mouse/src/main.c

```diff
@@ -228,8 +240,9 @@ void main(void)
 	}
 #endif
 
-	usb_hid_register_device(hid_report_desc, sizeof(hid_report_desc), NULL);
-	usb_hid_init();
+	usb_hid_register_device(hid_dev, hid_report_desc,
+				sizeof(hid_report_desc), NULL);
+	usb_hid_init(hid_dev);
```

Comment:
ACM CDC is different from the other classes since it has (serial) device already, all other classes should have device similar to HID Device.

About `usb_hid_init()`: if we move general init to `usb_init()` then the remaining part we can move to `usb_hid_register_device()`